lib/aes.ex

```diff
@@ -4,48 +4,59 @@ defmodule Fields.AES do
   https://en.wikipedia.org/wiki/Galois/Counter_Mode
   using a random Initialisation Vector for each encryption,
   this makes "bruteforce" decryption much more difficult.
+
   See `encrypt/1` and `decrypt/1` for more details.
   """
+
   # Use AES 256 Bit Keys for AES.
   @aad "AES256GCM"
 
   @doc """
   Uses a random IV for each call, and prepends the IV and Tag to the
-  ciphertext.  This means that `encrypt/1` will never return the same ciphertext
+  ciphertext. This means that `encrypt/1` will never return the same ciphertext
   for the same value. This makes "cracking" (bruteforce decryption) much harder!
+
   ## Parameters
   - `plaintext`: Accepts any data type as all values are converted to a String
-    using `to_string` before encryption.
+    using `to_string/1` before encryption.
+
   ## Examples
       iex> Fields.AES.encrypt("tea") != Fields.AES.encrypt("tea")
       true
+
       iex> ciphertext = Fields.AES.encrypt(123)
       iex> is_binary(ciphertext)
       true
   """
-  @spec encrypt(any) :: String.t
+  @spec encrypt(any) :: String.t()
   def encrypt(plaintext) do
-    iv = :crypto.strong_rand_bytes(16) # create random Initialisation Vector
-    {ciphertext, tag} = :crypto.block_encrypt(:aes_gcm, get_key(), iv, {@aad, to_string(plaintext), 16})
```

Comment:
I think it would be better if we supported both versions for a while. We can do that by using `Kernel.function_exported?`. So if the new function exists, we use it, otherwise we fallback to the old one.

What do you think?

Reply:
Can be a good idea, but it's necessary? The new function exists since OTP 22 https://www.erlang.org/doc/man/crypto.html#crypto_one_time_aead-6.

Of course, this will be a breaking change for apps that use OTP < 22, but they can also use older versions of the lib

Reply:
Ah okay, thought it was added on OTP 24. Should be fine then